toolchain/check/handle_class.cpp

```diff
@@ -581,17 +580,15 @@ static auto CheckCompleteAdapterClassType(Context& context,
     return SemIR::InstId::BuiltinError;
   }
 
-  if (auto fields = context.struct_type_fields().Get(fields_id);
-      !fields.empty()) {
-    auto [first_field_inst_id, _] = context.LookupNameInExactScope(
-        node_id, fields.front().name_id, class_info.scope_id,
-        context.name_scopes().Get(class_info.scope_id));
+  auto field_decls = context.field_decls_stack().PeekArray();
+  context.field_decls_stack().PopArray();
+  if (!field_decls.empty()) {
     CARBON_DIAGNOSTIC(AdaptWithFields, Error, "adapter with fields");
     CARBON_DIAGNOSTIC(AdaptWithFieldHere, Note,
                       "first field declaration is here");
     context.emitter()
         .Build(class_info.adapt_id, AdaptWithFields)
-        .Note(first_field_inst_id, AdaptWithFieldHere)
+        .Note(field_decls.front(), AdaptWithFieldHere)
```

Comment:
I think there might be a use-after-free here: `PeekArray` returns an `ArrayRef` pointing to the array and `PopArray` invalidates the storage for that array. Also, the return a couple of lines above would leave the fields array on the stack.

Would it make sense to sink both `PopArray` calls all the way down into a single call in `HandleParseNode(Context& context, Parse::ClassDefinitionId node_id)` after we call `CheckCompleteClassType`?